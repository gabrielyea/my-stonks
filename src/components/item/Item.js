/* eslint-disable react/prop-types */
import { motion } from 'framer-motion';
import { BiRightArrowCircle } from 'react-icons/bi';
import { Link } from 'react-router-dom';
import styles from './itemStyle.module.scss';

const variants = {
  initial: {
    y: 50,
    opacity: 0,
  },
  animate: {
    y: 0,
    opacity: 1,
  },
};

const child = {
  initial: {
    x: 50,
    opacity: 0,
  },
  animate: {
    x: 0,
    opacity: 1,
    transition: {
      staggerChildren: 0.2,
      delayChildren: 0.2,
    },
  },
};

const Item = ({ data }) => {
  const truncate = (str) => str.substring(0, 5);

  return (
    <motion.li
      className={styles.mainContainer}
      variants={variants}
    >
      <Link
        className={styles.link}
        to={`/detail/${data.id}`}
      >
        <BiRightArrowCircle className={styles.icon} />
      </Link>
      <h2>{data.symbol}</h2>
      <motion.div
        className={styles.dataContainer}
        variants={child}
        initial="initial"
        animate="animate"
      >
        <motion.p variants={child} className={styles.name}>{data.name}</motion.p>
        <motion.p variants={child}>{truncate(data.priceUsd)}</motion.p>
      </motion.div>
    </motion.li>
  );
};

export default Item;
