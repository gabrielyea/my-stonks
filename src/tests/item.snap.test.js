import React from 'react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Item from '../components/item/Item';

it('renders correctly when there is one item', () => {
  const data = {
    id: 1,
    name: 'BitCoin',
    priceUsd: '10',
    symbol: 'test',
  };
  const tree = renderer.create(
    <Router>
      <Item data={data} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
