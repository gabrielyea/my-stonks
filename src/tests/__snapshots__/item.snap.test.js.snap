// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is one item 1`] = `
<li
  className="mainContainer"
  style={Object {}}
>
  <a
    className="link"
    href="/detail/1"
    onClick={[Function]}
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.999,1.993c-5.514,0.001-10,4.487-10,10.001s4.486,10,10.001,10c5.513,0,9.999-4.486,10-10 C22,6.48,17.514,1.994,11.999,1.993z M12,19.994c-4.412,0-8.001-3.589-8.001-8s3.589-8,8-8.001C16.411,3.994,20,7.583,20,11.994 C19.999,16.405,16.41,19.994,12,19.994z"
      />
      <path
        d="M12 10.994L8 10.994 8 12.994 12 12.994 12 16 16.005 11.995 12 7.991z"
      />
    </svg>
  </a>
  <h2>
    test
  </h2>
  <div
    className="dataContainer"
    style={
      Object {
        "opacity": 0,
        "transform": "translateX(50px) translateZ(0)",
      }
    }
  >
    <p
      className="name"
      style={
        Object {
          "opacity": 0,
          "transform": "translateX(50px) translateZ(0)",
        }
      }
    >
      BitCoin
    </p>
    <p
      style={
        Object {
          "opacity": 0,
          "transform": "translateX(50px) translateZ(0)",
        }
      }
    >
      10
    </p>
  </div>
</li>
`;
